tests/index.js

```diff
@@ -186,6 +186,22 @@
     });
   });
 
+  var contentC = [
```

Comment:
This appears to pass even if the fix is removed. 
Without a reproduction of the problem, I don't really understand the fix here

Reply:
@stefanpenner The issue was that at [line 228](https://github.com/mukilane/layout-bin-packer/blob/45938deb17ece086b7f1bf3a6ebed3b0aeeacff0/lib/layout-bin-packer/shelf-first.js#L228), we divide by index, which is content.length. If suppose the content array is cleared, the subsequent calculations are affected, leading to `entry` being undefined and causing the #26 

This was reproduced in the test.

However, right now, after rebasing, I am getting stuck at an infinite loop at [246](https://github.com/mukilane/layout-bin-packer/blob/master/lib/layout-bin-packer/shelf-first.js#L246).

But the fix is to avoid the NaN.

Reply:
@mukilane oh ok, let's see if we can polish this up.

* when does length return something <= 0? a cleared array should be `length === 0` no?
* if we are guarding against `const averageHeight = bottom/index` being NaN, could we do so explicitly? 

```js
let averageHeight = bottom / index;
if (isNaN(averageHeight)) {
  averageHeight = 0; // or w/e value makes the most sense
}
```

Reply:
Actually, in that scenario. Wouldn't the `visibleStartingIndex` just be `0`?

Reply:
@stefanpenner I wonder if a non-empty list would return a height 0. 
Given that if length is 0, height would also be 0, so checking length would be fine, unless there is an odd case where entry's height is set to 0. :thinking: 

However, adding the height condition is fine with me, just in case.